doc/Dhcp_Mitigation/DHCP Mitigation.md

```diff
@@ -0,0 +1,203 @@
+-   It can also change or modify the data if needed. Traffic control(TC) uses queuing disciplines (qdiscs) to help organize and manage the transmission of traffic through a network interface. A qdisc performs two main functions:
+-       Enqueuing requests to place packets in a queue for later transmission.
+-       Dequeuing requests to select a packet from the queue for immediate transmission.
+
+To prevent a potential DHCP DoS attack on an interface, incoming traffic at the interface is rate-limited using traffic control(TC).
+
+### Proposed Mitigation
+
+DHCP traffic can be filtered and rate-limited by dropping all packets that exceed a user-specified rate, allowing legitimate users to be serviced by the DHCP server despite an ongoing attack. The design  provides a mechanism for DHCP rate-limiting on a specified interface.  Applying DHCP rate limit on a specific interface required two parameters:
+
+-   ##### Interface
+    The interface on which the DHCP rate limit is to be applied.
+
+
+-   ##### rate (packets per second)
+    An integer specifying a DHCP packet rate in packets per second. Since traffic control(TC) only supports rates in the form of bytes per second, this value is multiplied by 406 (number of bytes that make up a DHCP discover packet)
+
+Upon running this command, an ingress queuing discipline is created on the specified interface via traffic control(TC). Next, a traffic control(TC) filter is added to filter DHCP discover packets on protocol 17 (UDP) and destination port 67 (port used by DHCP), and a dropping action is applied to an incoming traffic rate. Incoming DHCP discover packets that exceed the rate are dropped to stop the attack from overwhelming the DHCP server.
+
+### Requirements
+
+- Support for Linux traffic control (tc) for implementing the rate-limiting mechanism 
+- Support for CLI commands for configuring DHCP rate-limiting
+
+### Architecture Design
+The overall SONiC architecture will remain the same and no new sub-modules will be introduced. Changes are made only in SONiC CLI where rate-limiting commands will be added that employ the Linux traffic control utility.
+
+
+
+### Sequence Diagram to Add Rate-limit
+
+
+![image](../../images/dhcp_mitigation_hld/add-seq.png)
+
+
+### Sequence Diagram to Del rate-limit 
+
+
+
+![image](../../images/dhcp_mitigation_hld/del-seq.png)
+
+
+### SAI API 
+
+- No SAI API change or addition is needed for this HLD. 
+
+
+### CLI/YANG model Enhancements 
+
+##### CLI  Configuration Commands
+
+Proposed SONiC CLI commands (sonic-utilities)
+-   config interface dhcp-mitigation-rate add [interface] [number of packets]
+-   config interface dhcp-mitigation-rate delete [interface] [number of packets]
+-   show interface dhcp-mitigation-rate
+
+Background Linux TC commands
+-   sudo tc qdisc add dev [Interface] handle ffff: ingress
```

Comment:
Current SONiC has integrated tc. If I understand correctly, this feature won't involve other dependencies or modules, only add cli in sonic-utilities?

Reply:
Yes, We are only proposing changing in CLI